sample/Scripts/CameraController.cpp

```diff
@@ -0,0 +1,117 @@
+#include "TestComponent.h"
+#include "Object/GameObject.h"
+#include "Object/Camera.h"
+#include "Core/Log.h"
+#include "Core/Time.h"
+#include "Math/Constants.h"
+#include "Input/InputSystem.h"
+
+#include "CameraController.h"
+
+CameraController::CameraController(gore::GameObject* gameObject) :
+    Component(gameObject),
+    m_Keyboard(nullptr),
+    m_Mouse(nullptr),
+    m_Yaw(0.0f),
+    m_Pitch(gore::math::constants::PI_4),
+    m_Roll(0.0f)
+{
+}
+
+CameraController::~CameraController()
+{
+}
+
+void CameraController::Start()
+{
+    const gore::InputSystem* inputSystem = gore::InputSystem::Get();
+
+    m_Keyboard = inputSystem->GetKeyboard();
+    m_Mouse    = inputSystem->GetMouse();
+
+    m_ActionMoveHorizontal = m_Keyboard->RegisterAction("MoveHorizontal", gore::KeyCode::D, gore::KeyCode::A);
+    m_ActionMoveVertical   = m_Keyboard->RegisterAction("MoveVertical", gore::KeyCode::W, gore::KeyCode::S);
+    m_ActionMoveUpDown     = m_Keyboard->RegisterAction("MoveUpDown", gore::KeyCode::E, gore::KeyCode::Q);
+
+    m_ActionAllowRotation = m_Mouse->RegisterAction("AllowRotation", gore::MouseButtonCode::Right);
+
+    m_ActionRotateHorizontal = m_Mouse->RegisterAction("RotateHorizontal", gore::MouseMovementCode::X);
+    m_ActionRotateVertical   = m_Mouse->RegisterAction("RotateVertical", gore::MouseMovementCode::Y);
+    m_ActionZoom             = m_Mouse->RegisterAction("Zoom", gore::MouseMovementCode::ScrollY);
+}
+
+void CameraController::Update()
+{
+    const float speed = 5.0f;
+
+    float deltaTime = GetDeltaTime();
+    float totalTime = GetTotalTime();
+
+    gore::Transform* transform = m_GameObject->GetComponent<gore::Transform>();
+
+    gore::Vector3 front = transform->GetLocalToWorldMatrix().GetForward();
+    gore::Vector3 right = transform->GetLocalToWorldMatrix().GetRight();
```

Comment:
Here it will be constructing two same matrices (if the compiler optimization is not smart enough)

But I think it could be fine here since the result of `GetLocalToWorldMatrix()` should be cached in a way other than changing caller's code.
Also performance is not what we are focusing on, at this time